pkg/util/inplaceupdate/adapter.go

```diff
@@ -74,7 +74,11 @@ type adapterInformer struct {
 }
 
 func (c *adapterInformer) getPod(namespace, name string) (*v1.Pod, error) {
```

Comment:
We usually do the DeepCopy in the caller side since not all getPod callers need to revise the object.

Reply:
There is a bug here. The adapterInfomer here is used by the UT and it behaves differently from the real informer which takes a deepCopy. This leads to UT can't repro the behavior in production.